lib/galaxy/managers/hdas.py

```diff
@@ -304,6 +304,8 @@ def __init__(self, app: StructuredApp):
                 "type",
                 "url",
                 "create_time",
+                "rerunnable",
+                "creating_job",
```

Comment:
These are expensive operations that I don't think belong in the summary view in the abstract.

```python
    @property
    def creating_job(self):
        # TODO this should work with `return self.dataset.job` (revise failing unit tests)
        creating_job_associations = None
        if self.creating_job_associations:
            creating_job_associations = self.creating_job_associations
        else:
            inherit_chain = self.source_dataset_chain
            if inherit_chain:
                creating_job_associations = inherit_chain[-1][0].creating_job_associations
        if creating_job_associations:
            return creating_job_associations[0].job
        return None
```

```python
    def serialize_rerunnable(self, item, key, **context):
        """
        Return False if this tool that created this dataset can't be re-run
        (e.g. upload).
        """
        dataset = item
        if dataset.creating_job:
            tool = self.app.toolbox.get_tool(dataset.creating_job.tool_id, dataset.creating_job.tool_version)
            if tool and tool.is_workflow_compatible:
                return True
        return False
```

If it is needed for the history panel, it is needed for the history panel and I can accept that - but if possible I would suggest maybe adding a "ui_internal" view or something to make this not part of the stable API?

Reply:
Thanks @jmchilton for looking into this. I moved these two attributes to the `detailed` view for now. It is likely that we will remove them even from there in a follow-up and just explicitly ask for those keys when forming the request url.

Reply:
Moving it back to `detailed` since that is where the current history expects it.